src/main/java/hudson/remoting/Util.java

```diff
@@ -247,4 +251,26 @@ static InetSocketAddress getResolvedHttpProxyAddress(String host, int port) thro
         }
         return targetAddress;
     }
+
+    @IgnoreJRERequirement @SuppressWarnings("Since15")
+    static void mkdirs(@Nonnull File file) throws IOException {
+        if (file.isDirectory()) return;
+
+        try {
+            Class.forName("java.nio.file.Files");
+            Files.createDirectories(file.toPath());
+            return;
+        } catch (ClassNotFoundException e) {
+            // JDK6
```

Comment:
I think we can finally drop support of Java 6 in remoting.
@stephenc is going to introduce some incompatible code, and maybe we will need remoting 3 in any case